rasp/selenium_engine.py

```diff
@@ -1,91 +1,31 @@
 import selenium
 import selenium.webdriver
-import time
 
 from rasp.base import Engine, Webpage
 from rasp.errors import EngineError
 
 
 class BaseSeleniumEngine(Engine):
-    def __init__(self):
-        self.driver = None
-        return
+    def __init__(self, *args, **kwargs):
+        self.driver = selenium.webdriver.Firefox()
 
     def get_source(self):
         return str(self.driver.page_source)
 
     def cleanup(self):
         self.driver.quit()
-        return
-
-    def setup(self):
-        self.driver = selenium.webdriver.Firefox()
-        return
 
     def load_page(self, url):
         self.driver.get(url)
-        return
 
     def get_url(self):
         return self.driver.current_url
 
-    def clone(self):
-        return BaseSeleniumEngine()
-
 
 class SeleniumEngine(BaseSeleniumEngine):
-    def __init__(self):
-        super(SeleniumEngine, self).__init__()
-        super(SeleniumEngine, self).setup()
-        return
-
     def get_page_source(self, url):
-        if url:
-            super(SeleniumEngine, self).load_page(url)
-            return Webpage(
-                    super(SeleniumEngine, self).get_url(),
-                    super(SeleniumEngine, self).get_source()
-                    )
-        else:
-            return None
-
-    def cleanup(self):
-        super(SeleniumEngine, self).cleanup()
-        return
-
-    def clone(self):
-        return SeleniumEngine()
-
-
-class TimedWaitEngine(BaseSeleniumEngine):
```

Comment:
we're going to move the timed behavior into the crawler class and out of (at least by default) from the engines.